File: Menu.js
const prompt = require('prompt-sync')();
const Card = require('./Card');
class Menu {
    constructor(wallet){
        this.wallet = wallet;
    }
    
     launch() {
            while (true) {
                console.log("1. Afficher le solde");
                console.log("2. Ajouter de l'argent");
                console.log("3. Retirer de l'argent");
                console.log("4. Afficher les cartes");
                console.log("5. Ajouter une carte");
                console.log("6. Retirer une carte");
                console.log("7. Ajouter des photos");
                console.log("8. Retirer des photos");
                console.log("9. Quitter");
                let choice = prompt("Choisissez une option : ");
                let amount, cardName, cardType, photoName, photoValue, description;
                switch (choice) {
                    case '1':
                       this.wallet.displayBalance();
                       break;
                    case '2':
                       amount = prompt("Montant à ajouter : ");
                       this.wallet.addMoney(amount);
                       break;
                    case '3':
                       amount = prompt("Montant à retirer : ");
                       this.wallet.withdrawMoney(amount);
                       break;
                       case '4':
                        this.wallet.displayAllCards();
                        break;
                    case '5':
                        cardName = prompt("Nom de la carte : ");
                        cardType = prompt("Type de la carte : ");
                        const newCard = new Card(cardName, cardType);
                        this.wallet.addCard(newCard);
                        console.log("Carte ajoutée : " + newCard.getName());
                        break;
                    case '6':
                        cardName = prompt("Nom de la carte à retirer : ");
                        this.wallet.removeCard(cardName);
                        break;
                     case '7':
                        console.log("Pas eencore fonctonnel");
                     //       photoName = prompt("Nom de la photo : ");
                     //       photoValue = parseInt(prompt("Valeur de la photo : "));
                     //       const newPhoto = new Photos();
                     //       newPhoto.setName(photoName);
                     //       newPhoto.setValues(photoValue);
                     //       this.wallet.addPhotos(newPhoto);
                     //       console.log("Photo ajoutée : " + newPhoto.getName());
                     //       break;
                      case '8':
                        console.log("Pas encore fonctionnel");
                           // photoName = prompt("Nom de la photo à retirer : ");
                           // this.wallet.removePhotos(photoName);
                           // break;
                       case '9':
                        console.log("Au revoir !");   
                        return
                    default:
                       console.log("Choix invalide, veuillez réessayer.");
                }
            }
        }
}

module.exports = Menu

File: menu.py
from wallet import Wallet
from card import Card

class Menu:
    def __init__(self, wallet):
        self.wallet = wallet

    def menu(self):
        while True:
            print("\n1. Afficher le solde")
            print("2. Ajouter de l'argent")
            print("3. Retirer de l'argent")
            print("4. Afficher les cartes")
            print("5. Ajouter une carte")
            print("6. Retirer une carte")
            print("7. Ajouter des photos")
            print("8. Retirer des photos")
            print("9. Quitter")
            choice = input("Veuillez choisir une option : ")
            amount, card_name, card_type, photo_name, photo_value, description = 0, "", "", "", 0, ""

            if choice == "1":
                self.wallet.display_balance()
            elif choice == "2":
                description = input("Description : ")
                amount = float(input("Veuillez entrer le montant à ajouter : "))
                self.wallet.add_money(amount)
            elif choice == "3":
                description = input("Description : ")
                amount = float(input("Veuillez entrer le montant à retirer : "))
                self.wallet.withdraw_money(amount)
            elif choice == '4':
                self.wallet.display_all_cards()
            elif choice == '5':
                card_name = input("Nom de la carte : ")
                card_type = input("Type de la carte : ")
                new_card = Card(card_name, card_type)
                self.wallet.add_card(new_card)
                print("Carte ajoutée :", new_card.get_name())
            elif choice == '6':
                card_name = input("Nom de la carte à retirer : ")
                self.wallet.remove_card(card_name)
            elif choice == '7':
                # photo_name = input("Nom de la photo : ")
                # photo_value = int(input("Valeur de la photo : "))
                # new_photo = Photos()
                # new_photo.set_name(photo_name)
                # new_photo.set_values(photo_value)
                # self.wallet.add_photos(new_photo)
                # print("Photo ajoutée :", new_photo.get_name())
                print("Pas encore fonctionnel")
            elif choice == '8':
                # photo_name = input("Nom de la photo à retirer : ")
                # self.wallet.remove_photos(photo_name)
                print("Pas encore fonctionnel")
            elif choice == '9':
                print("Au revoir !")
                return
            else:
                print("Choix invalide, veuillez réessayer.")
